passes/cmds/internal_stats.cc

```diff
@@ -0,0 +1,120 @@
+/*
+ *  yosys -- Yosys Open SYnthesis Suite
+ *
+ *  Copyright (C) 2012  Claire Xenia Wolf <claire@yosyshq.com>
+ *
+ *  Permission to use, copy, modify, and/or distribute this software for any
+ *  purpose with or without fee is hereby granted, provided that the above
+ *  copyright notice and this permission notice appear in all copies.
+ *
+ *  THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+ *  WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+ *  MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+ *  ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+ *  WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+ *  ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+ *  OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+ *
+ */
+
+#include <iterator>
+#include <optional>
+#include <stdint.h>
+
+#include "kernel/yosys.h"
+#include "kernel/celltypes.h"
+#include "passes/techmap/libparse.h"
+#include "kernel/cost.h"
+#include "libs/json11/json11.hpp"
+
+#if defined(__APPLE__) && defined(__MACH__)
+#include <mach/task.h>
+#include <mach/mach_init.h>
+#endif
+
+USING_YOSYS_NAMESPACE
+PRIVATE_NAMESPACE_BEGIN
+
+std::optional<uint64_t> current_mem_bytes() {
+
+#if defined(__APPLE__)
+    task_basic_info_64_data_t basicInfo;
+    mach_msg_type_number_t count = TASK_BASIC_INFO_64_COUNT;
+    kern_return_t error = task_info(mach_task_self(), TASK_BASIC_INFO_64, (task_info_t)&basicInfo, &count);
+    if (error != KERN_SUCCESS) {
+        return {}; // Error getting task information
+    }
+    return basicInfo.resident_size; // Return RSS in KB
+
+#elif defined(__linux__)
+	// Not all linux distributions have to have this file
+    std::ifstream statusFile("/proc/self/status");
+    std::string line;
+    while (std::getline(statusFile, line)) {
+        if (line.find("VmRSS:") == 0) {
+            std::istringstream iss(line);
+            std::string token;
+			// Skip prefix
+            iss >> token;
+            uint64_t rss;
+            iss >> rss;
+            return rss * 1024;
+        }
+    }
+	// Error reading /proc/self/status
+    return {};
+
+#else
+	return {};
+#endif
+}
+
+struct InternalStatsPass : public Pass {
+	InternalStatsPass() : Pass("internal_stats", "print some internal statistics") { }
+	void help() override
+	{
+		//   |---v---|---v---|---v---|---v---|---v---|---v---|---v---|---v---|---v---|---v---|
+		log("\n"); // TODO
```

Comment:
Help missing, not something we should let into a release

Reply:
TIL empty help messages causes docs to fail (granted it is technically a warning, but warnings are errors in our ReadtheDocs setup)
`/home/docs/checkouts/readthedocs.org/user_builds/yosys/checkouts/latest/docs/source/cmd/internal_stats.rst:21: WARNING: Literal block expected; none found. [docutils]`